libp2p-networking/src/network/mod.rs

```diff
@@ -223,32 +222,72 @@ pub async fn gen_transport(
     }
     .await?;
 
-    // keys for signing messages
-    let noise_keys = noise::Keypair::<noise::X25519Spec>::new()
-        .into_authentic(&identity)
-        .expect("Signing libp2p-noise static DH keypair failed.");
     let multiplexing_config = {
         let mut yamux_config = YamuxConfig::default();
         yamux_config.set_window_update_mode(WindowUpdateMode::on_read());
-
         yamux_config
-            .map_inbound(StreamMuxerBox::new)
-            .map_outbound(StreamMuxerBox::new)
     };
 
     Ok(transport
-        .upgrade(upgrade::Version::V1)
-        // authentication: messages are signed
-        .authenticate(noise::NoiseConfig::xx(noise_keys).into_authenticated())
-        // muxxing streams
-        // useful because only one connection opened
-        // with multiple substreams
-        // https://docs.libp2p.io/concepts/stream-multiplexing/
+        .upgrade(Version::V1)
+        .authenticate(libp2p_noise::Config::new(&identity).unwrap())
```

Comment:
This is interesting. So the authentication model for this changed?

Reply:
I don't think it necessarily changed. Rather, XX became the default and they deprecated the other handshakes. I was manually specifying XX because I was afraid the default would change and I wouldn't notice. https://github.com/libp2p/rust-libp2p/blob/master/transports/noise/CHANGELOG.md